### PR TITLE
correct nav bar resizing and display

### PR DESCRIPTION
**Pull request by kbhawkey**
Revert additions from pull request 18888.
(Testing tablet size screen and drop down menus in header)

### CODE REVIEW

**Comment by kbhawkey** [comment]
For further information, see issue https://github.com/kubernetes/website/issues/18972.

**Comment by sftim** [comment]
@kbhawkey, can you check the description for this PR?

I'm not sure whether or not this is still a work-in-progress.

**Comment by kbhawkey** [comment]
> 
> 
> @kbhawkey, can you check the description for this PR?
> 
> I'm not sure whether or not this is still a work-in-progress.

You don't like the word `testing` 😄 ? I can change the description.
There are several problems going on. I am not sure if the Safari related problem is still an issue.
With this change the `hamburger` displays for very small screens and the nav bar displays for medium to largish desktop screens (at least in my testing). There is room for improvement.
I consider these changes a temporary fix -- at least until the new theme/layout is in place.

**Comment by sftim** [comment]
At 1000px wide, the deploy preview looks different to the live site; I think that's a regression (the Kubernetes logo appears behind the navigation text).

I'm using Firefox 73.0.1

Kubernetes Documentation preview

**Comment by kbhawkey** [comment]
@sftim , I agree with your assessment. I'll take another look.

**Comment by kbhawkey** [comment]
Here is what I found:
With the changes from PR #18888, for screens > 800px && < 1025px, the window (that is toggled when the menu is clicked) no longer displays content (mainNav).

What to do:
- Clean up the changes from the previous commit (and adjust this PR). Screens > 800px && < 1025 will continue to display content when menu is selected (though not the full nav bar).
- What was the original issue with Safari browser and drop down menus on nav bar. I see that the drop down menus are not accessible from the original issue (@burningalchemist can you provide more details, screen size)?
- Could adjust the js so that the vertical nav bar is displayed (push menu) for tablet screens?

I am leaning toward getting the UI back to a working state and go from there.

**Comment by sftim** [comment]
> I am leaning toward getting the UI back to a working state and go from there.

:+1:

**Comment by burningalchemist** [comment]
@kbhawkey Yes, you're right, I can see a missing section for a tablet screen.

The fix for dropdown lists was tested on mobiles, laptops and larger displays. I've also asked if `mainNav` is obsolete as I couldn't find it anyhow (I didn't have a tablet to test it).

The content of `mainNav` actually seems obsolete in my opinion. So I'd rather apply the same `hamburger` menu as for mobile version, but it's up to the maintainers.

**Comment by kbhawkey** [comment]
> 
> 
> @kbhawkey Yes, you're right, I can see a missing section for a tablet screen.
> 
> The fix for dropdown lists was tested on mobiles, laptops and larger displays. I've also asked if 

I would expect the mobile screen to use the vertical list for the nav bar.

`mainNav` is obsolete as I couldn't find it anyhow (I didn't have a tablet to test it).

When the window is toggled open (I don't have a tablet. You could minimize a desktop screen to tablet size), view the page source.

> 
> The content of `mainNav` actually seems obsolete in my opinion. So I'd rather apply the same `hamburger` menu as for mobile version, but it's up to the maintainers.

I think it makes sense to offer the full navigation for all screens.

@burningalchemist , is there a remaining issue with drop down menu, Safari browser desktop or mobile?

**Comment by burningalchemist** [comment]
@kbhawkey yeah, I was able to find the issue now by sizing the window, thanks. :)

It's Safari desktop, as Safari mobile is using a vertical hamburger menu. If I use a preview link from the PR, the issue is back. Also previous versions like `v.1.16` have a dropdown list issue as they're not updated anymore.

>I think it makes sense to offer the full navigation for all screens.
Yeah, I agree. I mean, `mainNav` seems to be poor on options (can't choose a version or a language from there), that's why it was concerned obsolete and not used. It would be cool to enable these features for tablets as well.

File: assets/sass/_base.sass [hunk 1]
```diff
@@ -244,8 +245,7 @@ header
       background-color: white
 
 #mainNav
```
Comment (sftim) [inline assets/sass/_base.sass]: Also drop this line?

Comment (burningalchemist) [reply assets/sass/_base.sass]: @sftim this one should remain as further attributes belong to the content within `#mainNav` nav element.

Comment (burningalchemist) [reply assets/sass/_base.sass]: @sftim this one should remain as further attributes belong to the content within `#mainNav` nav element.

Comment (sftim) [reply assets/sass/_base.sass]: Oh, it's Sass and I hadn't noticed. D'oh!

**Comment by kbhawkey** [comment]
> 
> 
> @kbhawkey @sftim once the commit is reverted, I can take another shot to fix dropdown lists in Safari (keeping `#mainNav` in mind). They seem to be broken for a long time. :)

@burningalchemist, Is there a specific page that you used to test, or did you see the issue with the home page (+ Safari browser + desktop)?

**Comment by burningalchemist** [comment]
@kbhawkey, the block with dropdown lists comes from `layouts/partials/header.html`, so it exists on every page for a desktop version of the website in Safari.

**Comment by kbhawkey** [comment]
@sftim, @jimangel  . This pull request would reset the state of the navigation bar by removing the Safari related changes from PR #18888. If it is better to `revert` the previous commit, this could also be done. I think this change should be made so that the UI is back to a working condition (Yes, the Safari issues remain). Would you have time to review?

**Comment by brianpursley** [comment]
Preview looks good to me.

Hamburger works fine now (in the preview) on iPad safari and chrome.  Current live site does not.

/lgtm

**Comment by kbhawkey** [comment]
> I visited https://deploy-preview-19477--kubernetes-io-master-staging.netlify.com/docs/home/ and set a narrow viewport. The top-left hamburgerish menu doesn't do anything on that page, in Firefox and Chrome. This is _not_ a regression, I see the same on the live site.
> 
> I haven't got familiar with the CSS. I'm reasonably confident that these changes aren't making thing worse. I'm happy to see these changes merged.
> /lgtm

@sftim, The changes are reverting the content back to a previous state (problem outlined in issue #18972). I did not address the larger problems (hamburger) or fix the problems with the UI and Safari desktop.
No additional fixes or changes were added -- just discussed.
This PR could be changed to just revert PR #18888.

**Pull request merged**